mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java

```diff
@@ -169,7 +173,7 @@ protected void onRemove() {
 
     final synchronized void assign(Redrawer redrawer) {
         if (this.assignedRedrawer != null) {
-            throw new IllegalStateException("layer already assigned");
+            LOGGER.warning("layer already assigned");
```

Comment:
Probably it is better to `return;` after the warning?

Reply:
wasn't sure about what is more important - `onAdd()` getting called or directly exiting on invalid state - but added `return` now